pyccel/parser/syntax/headers.py

```diff
@@ -157,6 +157,38 @@ def expr(self):
             d_var['order'] = order
         return d_var
 
+class MacroResult(BasicStmt):
+    """class representing a MacroResult in the grammar."""
+
+    def __init__(self, **kwargs):
+        """Constructor for MacroResult
+
+        name: str
+            Name of the variable result
+
+        shape: list
+            a list representing the shape of the result
+        """
```

Comment:
`__init__` functions do not need doc strings. In order to call a constructor you do:
```
MacroResult(..)
```
Therefore in order to see the docstring you would usually do;
```
MacroResult?
```
This shows the docstring for the class which is why the arguments for the `__init__` function should go in the class docstring

Reply:
Fixed here 0717085